### DISCUSSION TITLE
Consistency in navigation

### DISCUSSION THREAD

**Discussion by goyalyashpal** [helix-editor/helix#5299] [Ideas] [answered]
- i like `hjkl` navigation,

* but using `C-n` or `C-p`, 
* this is such a turn off in any other mode like in "file picker" `space f`, "references" `g r`, "symbol picker" `space s`, "diagnostic picker" `space G` (or even in other softwares)

`C-d` or `C-u`  is also used in some places https://github.com/helix-editor/helix/discussions/5923#discussioncomment-4946453

This was also mentioned briefly by Luke Pighetti

> we're using `control p` and `control n` to go up and down here.
Because you can imagine if you `hjkl` it would go in here [input field].
So you need a different key, that's what `control p` and `control n` are for
>
> \- [video] _"Helix – Getting Dangerous"_ @ 5:03 by Luke Pighetti

[video]: https://www.youtube.com/watch?v=i_qsYALaFVo&t=5m03s "video: Helix – Getting Dangerous @5:03 `g r`"

can something be done about this? i really would like to see the "modal" approach in navigation in those screens other than editor as well.

I am on version 22.05

**Comment by Omnikar** [comment]
I usually use tab and shift-tab to navigate those menus.

**Reply by goyalyashpal** [reply]
that's a bit better, but ....

**Comment by vasylnakvasiuk** [comment, marked as answer]
For me `C-j` and `C-k` work much better (easy to use, your finger always on the home position and also [Telescope.nvim](https://github.com/nvim-telescope/telescope.nvim) uses this approach). And I have an issue for this https://github.com/helix-editor/helix/issues/5232.
If I understood correctly, there was one argument against: `C-k` conflicted with the basic `readline` mapping (Delete everything after cursor), but I don't know how often we meet this use case.

So, `C-j`/`C-k` were added by https://github.com/helix-editor/helix/pull/876 and then were deleted by https://github.com/helix-editor/helix/pull/1792. And there's an open issue for customizing picker keys which would allow adding what you want https://github.com/helix-editor/helix/issues/2757.

But my vote 👍  for `C-j`/`C-k` in pickers by default.

**Reply by goyalyashpal** [reply]
thanks for detailed answer. nice to know their name (called picker keys).

i haven't used helix much, so, don't know about readline mapping, or such.

**Reply by goyalyashpal** [reply]
#5505 <-- **2757** <-- 2766 <--  **5232**
